bots/pr/src/main/java/org/openjdk/skara/bots/pr/PullRequestInstance.java

```diff
@@ -133,16 +133,31 @@ private Hash commitSquashed(List<Review> activeReviews, Namespace namespace, Str
     }
 
     private Hash commitMerge(List<Review> activeReviews, Namespace namespace, String censusDomain) throws IOException {
-        localRepo.checkout(headHash, true);
+        // Find the last merge commit - the very last commit is not eligible (as the merge needs a parent)
+        var commits = localRepo.commits(baseHash + ".." + headHash).asList();
+        int mergeCommitIndex = commits.size();
+        for (int i = 0; i < commits.size() - 1; ++i) {
+            if (commits.get(i).isMerge()) {
+                mergeCommitIndex = i;
+            }
+        }
 
         var contributor = namespace.get(pr.author().id());
         if (contributor == null) {
             throw new RuntimeException("Merges can only be performed by Committers");
         }
 
         var author = new Author(contributor.fullName().orElseThrow(), contributor.username() + "@" + censusDomain);
-
         var commitMessage = commitMessage(activeReviews, namespace, true);
+
+        localRepo.checkout(commits.get(mergeCommitIndex).hash(), true);
```

Comment:
Check that `mergeCommitIndex < commits.size()` before and otherwise throw an exception?

Reply:
That should never happen (tm), the actual merge depends on jcheck passing first so we can assume that the PR is well-formed.

Reply:
Ok, sounds good!